framework/src/modules/dpos/transaction_assets/register_transaction_asset.ts

```diff
@@ -0,0 +1,167 @@
+import { ValidationError } from '../../../errors';
+import { CHAIN_STATE_DELEGATE_USERNAMES, DELEGATE_NAME_FEE } from '../constants';
+
+const isNullCharacterIncluded = (input: string): boolean =>
+	new RegExp(/\\0|\\u0000|\\x00/).test(input);
+
+const isUsername = (username: string): boolean => {
+	if (isNullCharacterIncluded(username)) {
+		return false;
+	}
+
+	if (username !== username.trim().toLowerCase()) {
+		return false;
+	}
+
+	if (/^[0-9]{1,21}[L|l]$/g.test(username)) {
+		return false;
+	}
+
+	return /^[a-z0-9!@$&_.]+$/g.test(username);
+};
+
+const delegatesUserNamesSchema = {
+	$id: '/dpos/userNames',
+	type: 'object',
+	properties: {
+		registeredDelegates: {
+			type: 'array',
+			fieldNumber: 1,
+			items: {
+				type: 'object',
+				required: ['username', 'address'],
+				properties: {
+					username: {
+						dataType: 'string',
+						fieldNumber: 1,
+					},
+					address: {
+						dataType: 'bytes',
+						fieldNumber: 2,
+					},
+				},
+			},
+		},
+	},
+	required: ['registeredDelegates'],
+};
+
+interface RegisteredDelegate {
+	readonly username: string;
+	readonly address: Buffer;
+}
+
+interface RegisteredDelegates {
+	registeredDelegates: RegisteredDelegate[];
+}
+interface DelegatePersistedUsernames {
+	readonly registeredDelegates: RegisteredDelegate[];
+}
+
+export interface RegisterTransactionAssetInput {
+	readonly username: string;
+}
+
+const getRegisteredDelegates = async (store: StateStore): Promise<DelegatePersistedUsernames> => {
+	const usernamesBuffer = await store.chain.get(CHAIN_STATE_DELEGATE_USERNAMES);
+	if (!usernamesBuffer) {
+		return { registeredDelegates: [] };
+	}
+	const parsedUsernames = codec.decode<RegisteredDelegates>(
+		delegatesUserNamesSchema,
+		usernamesBuffer,
+	);
+
+	parsedUsernames.registeredDelegates = parsedUsernames.registeredDelegates.map(
+		(value: { address: Buffer; username: string }) => ({
+			username: value.username,
+			address: value.address,
+		}),
+	);
+
+	return parsedUsernames as DelegatePersistedUsernames;
+};
+
+export class RegisterTransactionAsset extends BaseAsset<RegisterTransactionAssetInput> {
+	public baseFee = DELEGATE_NAME_FEE;
+	public name = 'register';
+	public type = 0;
+	public assetSchema = {
+		$id: 'lisk/dpos/register',
+		type: 'object',
+		required: ['username'],
+		properties: {
+			username: {
+				dataType: 'string',
+				fieldNumber: 1,
+				minLength: 1,
+				maxLength: 20,
+			},
+		},
+	};
+
+	// eslint-disable-next-line class-methods-use-this
+	public validateAsset({ asset }: ValidateAssetInput<RegisterTransactionAssetInput>): void {
+		if (!isUsername(asset.username)) {
+			throw new ValidationError('The username is in unsupported format', asset.username);
+		}
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	public async applyAsset({
+		asset,
+		senderID,
+		stateStore,
+	}: ApplyAssetInput<RegisterTransactionAssetInput>): Promise<void> {
+		const sender = await stateStore.account.get<{
```

Comment:
maybe refactoring but account type for this module can be defined in somewhere common within module

Reply:
In coming PR I will extract it to a module level and reuse it.